ValveResourceFormat/Resource/ResourceTypes/ChoreoDataList.cs

```diff
@@ -0,0 +1,143 @@
+    {
+        public int Version { get; private set; }
+        public ChoreoScene[] Scenes { get; private set; }
+        public override void Read(BinaryReader reader, Resource resource)
+        {
+            reader.BaseStream.Position = Offset;
+
+            Version = reader.ReadInt32();
+            var sceneCount = reader.ReadInt32();
+            var strings = ReadStrings(reader);
+            reader.ReadInt32(); //Proportional to total length of strings or file size. Probably nothing useful.
+
+            Scenes = ReadScenes(reader, sceneCount, strings);
+        }
+
+        public override void WriteText(IndentedTextWriter writer)
+        {
+            foreach (var item in Scenes)
+            {
+                writer.WriteLine(item.Name);
+            }
+        }
+
+        private static long ReadPosition(BinaryReader reader)
+        {
+            return reader.BaseStream.Position + reader.ReadInt32();
+        }
+
+
+        private static ChoreoScene[] ReadScenes(BinaryReader reader, int sceneCount, string[] strings)
+        {
+            var scenes = new ChoreoScene[sceneCount];
+            for (var i = 0; i < sceneCount; i++)
+            {
+                scenes[i] = ReadScene(reader, strings);
+            }
+            return scenes;
+        }
+
+        private static ChoreoScene ReadScene(BinaryReader reader, string[] strings)
+        {
+            var namePosition = ReadPosition(reader);
+            var blockPosition = ReadPosition(reader);
+            var length = reader.ReadInt32();
+            var sceneDuration = reader.ReadInt32();
+            var sceneSoundDuration = reader.ReadInt32();
+            var hasSounds = reader.ReadInt32(); //This is always 0 or 1
+
+            var previousPosition = reader.BaseStream.Position;
+
+            reader.BaseStream.Position = namePosition;
+            var name = reader.ReadNullTermString(Encoding.ASCII);
+
+            reader.BaseStream.Position = blockPosition;
+            var sceneBlock = ReadSceneBlock(reader, length);
+
+            reader.BaseStream.Position = previousPosition;
+
+            using var sceneStream = new MemoryStream(sceneBlock);
+            var scene = BVCDParser.Parse(sceneStream, strings);
+            scene.Name = name;
+            scene.Duration = sceneDuration;
+            scene.SoundDuration = sceneSoundDuration;
+            scene.HasSounds = hasSounds != 0;
+
+            return scene;
+        }
+
+        private static byte[] ReadSceneBlock(BinaryReader reader, int length)
+        {
+            var id = new string(reader.ReadChars(4));
+            if (id == "LZMA")
+            {
+                var uncompressedLength = reader.ReadInt32();
+                var compressedLength = reader.ReadInt32();
+
+                var lzmaDecoder = new LzmaDecoder();
+                lzmaDecoder.SetDecoderProperties(reader.ReadBytes(5));
+                var compressedBuffer = reader.ReadBytes(compressedLength);
```

Comment:
Use ArrayPool

Reply:
I've removed the compressedBuffer variable entirely, I'm guessing that's where you meant it. I was considering returning an arraypool array too, but that'd probably be a bit awkward to handle for functions that call this.

Reply:
Yep i meant compressed buffer, using the stream directly is also good.